File: spec/java/src/io/kaitai/struct/spec/TestExprBytesCmp.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.ExprBytesCmp;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestExprBytesCmp extends CommonSpec {
    @Test
    public void testExprBytesCmp() throws Exception {
        ExprBytesCmp r = ExprBytesCmp.fromFile(SRC_DIR + "fixed_struct.bin");

        assertEquals(r.one(), "P".getBytes());
        assertEquals(r.two(), "ACK".getBytes());

        assertEquals(r.isEq().booleanValue(), true);
        assertEquals(r.isNe().booleanValue(), false);
        assertEquals(r.isLt().booleanValue(), true);
        assertEquals(r.isGt().booleanValue(), false);
        assertEquals(r.isLe().booleanValue(), true);
        assertEquals(r.isGe().booleanValue(), false);
        assertEquals(r.isLt2().booleanValue(), false);
        assertEquals(r.isGt2().booleanValue(), true);
    }
}

File: spec/javascript/test_expr_bytes_cmp.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('ExprBytesCmp', 'src/fixed_struct.bin', function(r) {
  assert.equal(KaitaiStream.bytesToStr(r.one, "UTF-8"), "P");
  assert.equal(KaitaiStream.bytesToStr(r.two, "UTF-8"), "ACK");

  assert.equal(r.isEq, true);
  assert.equal(r.isNe, false);
  assert.equal(r.isLt, true);
  assert.equal(r.isGt, false);
  assert.equal(r.isLe, true);
  assert.equal(r.isGe, false);
  assert.equal(r.isLt2, false);
  assert.equal(r.isGt2, true);
});
